src/Microsoft.SqlTools.ServiceLayer/SchemaDesigner/SchemaDesignerService.cs

```diff
@@ -54,80 +54,85 @@ await requestContext.SendResult(new CreateSessionResponse()
+                        sessions.Remove(requestParams.SessionId);
+                        SchemaDesignerQueryExecution.Disconnect(requestParams.SessionId);
+                    }
                 }
-            }
-            catch (Exception e)
-            {
-                Logger.Error(e.Message);
-            }
-            await requestContext.SendResult(new DisposeSessionResponse());
+                catch (Exception e)
+                {
+                    Logger.Error(e.Message);
+                }
+                await requestContext.SendResult(new DisposeSessionResponse());
+            });
         }
 
-        internal async Task HandlePublishSchemaDesignerSessionRequest(PublishSessionRequest requestParams, RequestContext<PublishSessionResponse> requestContext)
+        internal Task HandlePublishSchemaDesignerSessionRequest(PublishSessionRequest requestParams, RequestContext<PublishSessionResponse> requestContext)
         {
-            try
+            return this.HandleRequest<PublishSessionResponse>(requestContext, async () =>
             {
                 if (sessions.TryGetValue(requestParams.SessionId, out SchemaDesignerSession? session))
                 {
                     session.PublishSchema();
                 }
-            }
-            catch (Exception e)
-            {
-                Logger.Error(e.Message);
-                await requestContext.SendError(e);
-            }
-            await requestContext.SendResult(new PublishSessionResponse());
+                await requestContext.SendResult(new PublishSessionResponse());
+            });
         }
 
 
-        internal async Task HandleGetSchemaDesignerSessionReportRequest(GetReportRequest requestParams, RequestContext<GetReportResponse> requestContext)
+        internal Task HandleGetSchemaDesignerSessionReportRequest(GetReportRequest requestParams, RequestContext<GetReportResponse> requestContext)
         {
-            SchemaDesignerSession session = sessions[requestParams.SessionId];
-            var report = await session.GetReport(requestParams.UpdatedSchema);
-            try
+            return this.HandleRequest<GetReportResponse>(requestContext, async () =>
             {
+                SchemaDesignerSession session = sessions[requestParams.SessionId];
+                var report = await session.GetReport(requestParams.UpdatedSchema);
                 await requestContext.SendResult(report);
-            }
-            catch (Exception e)
+            });
+        }
+
+        private Task HandleRequest<T>(RequestContext<T> requestContext, Func<Task> action)
+        {
+            // The request handling will take some time to return, we need to use a separate task to run the request handler so that it won't block the main thread.
+            // For any specific table designer instance, ADS UI can make sure there are at most one request being processed at any given time, so we don't have to worry about race conditions.
```

Comment:
Update comment (ADS, Table Designer).  Also, can this instead be moved someplace else so that it's shared with Table Designer instead of copied?